File: src/main/kotlin/dev/jefy/connectpro/user/applicaion/dtos/ValidateResetPasswordOtpRequest.java
package dev.jefy.connectpro.user.applicaion.dtos;


import org.jspecify.annotations.NullMarked;

import java.util.UUID;

/**
 * @author Jôph Yamba
 */
@NullMarked
public record ValidateResetPasswordOtpRequest(UUID tokenId, int code) {}

File: src/main/kotlin/dev/jefy/connectpro/user/application/dtos/ValidateResetPasswordOtpRequest.kt
package dev.jefy.connectpro.user.application.dtos

import jakarta.validation.constraints.NotBlank
import org.hibernate.validator.constraints.Length
import java.util.*

data class ValidateResetPasswordOtpRequest(
    val tokenId: UUID,
    @field:NotBlank(message = "OTP must not be blank")
    @field:Length(min = 4, max = 4, message = "OTP must be 4 digits")
    val code: String
)